File: Iniciante/URI 1010.pas
var
    cod1 : int64;
    cod2 : int64;
    n1 : int64;
    n2 : int64;
	valor1 : double;
	valor2 : double;
	total : double;

begin
    readLn(cod1, n1, valor1);
    readLn(cod2, n2, valor2);
	
    total := (n1 * valor1) + (n2 * valor2);
	
    writeln('VALOR A PAGAR: R$ ', total:0:2);
end.

File: Iniciante/URI 1010.hs
import Text.Printf

main = do
    line1 <- getLine
    line2 <- getLine

    let (cod1:n1:valor1:_) = map (read :: String -> Double) $ words line1
    let (cod2:n2:valor2:_) = map (read :: String -> Double) $ words line2
    let total = (n1 * valor1) + (n2 * valor2)

    putStrLn $ printf "VALOR A PAGAR: R$ %.2f" total